simulariumio/custom_converter.py

```diff
@@ -92,3 +118,263 @@ def _read(self, input_data: CustomData) -> Dict[str, Any]:
             "data": input_data.plots,
         }
         return simularium_data
+
+    @staticmethod
+    def _get_spatial_bundle_data_subpoints(
+        agent_data: AgentData,
+    ) -> List[Dict[str, Any]]:
+        """
+        Return the spatialData's bundleData for a simulation
+        of agents with subpoints, packing buffer with jagged data is slower
```

Comment:
I have often wondered if it would be more efficient to keep the jagged data separate.  Put subpoints in a different buffer or have them entirely at the end of the per-agent data.   It's not only faster to write to, but also faster to parse if the data is a fixed known size per agent.  Just thinking out loud (this gives me an idea to have a simplified fiber mode that doesn't need to create the full 3d tubes).

Reply:
Yeah I've wondered that too and I think we've discussed it a little in the past, but definitely could discuss again